cpp/dolfinx/fem/dofmapbuilder.cpp

```diff
@@ -275,26 +282,25 @@ build_basic_dofmap(const mesh::Topology& topology,
 /// @param [in] dofmap The basic dofmap data
 /// @param [in] dof_entity Map from dof index to (dim, entity_index),
 /// where entity_index is the process-wise mesh entity index
-/// @param [in] topology The mesh topology
+/// @param [in] index_maps The set of index maps referring to dofs in dof_entity
 /// @param [in] reorder_fn Graph reordering function that is applied for
 /// dof re-ordering
 /// @return The pair (old-to-new local index map, M), where M is the
 /// number of dofs owned by this process
 std::pair<std::vector<std::int32_t>, std::int32_t> compute_reordering_map(
-    mdspan2_t<const std::int32_t> dofmap,
+    const std::vector<mdspan2_t<const std::int32_t>>& dofmaps,
     const std::vector<std::pair<std::int8_t, std::int32_t>>& dof_entity,
-    const mesh::Topology& topology,
+    const std::vector<std::shared_ptr<const common::IndexMap>>& index_maps,
```

Comment:
I'm keeping this for now. (a) it is an internal namespaced function, (b) as a free function, it's not holding anything, but mostly, because I currently need to check the validity with `if (map)`. I dont' think `reference_wrapper` can wrap `nullptr`.